### PR TITLE
No polling

### PR DESCRIPTION
**Pull request by roberttod**
- Tasks are processed by baton passing rather than polling
- [x] Functional tests working
- [x] Functional test for task timeout
- [x] Unit tests updated

### CODE REVIEW

**Comment by swagatha-christie** [comment]
By analyzing the blame information on this pull request, we identified @davidpadbury to be a potential reviewer

File: test/functional/test-queue-that-functional.js [hunk 1]
```diff
@@ -1,11 +1,10 @@
 /* global describe, it, expect, beforeEach, afterEach, sinon */
 var _ = require('underscore')
 var createQueueThat = require('../../lib/queue-that')
-describe('queueThat (functional)', function () {
+describe.only('queueThat (functional)', function () {
```
Comment (davidpadbury) [inline test/functional/test-queue-that-functional.js]: is that intentional?

**Comment by davidpadbury** [comment]
It's looking good. The flow feels like it's getting much harder to understand, but re-reading the full thing still makes sense. Perhaps it's worth writing some very brief docs at some point just going however how queue management works.

File: test/functional/test-queue-that-functional.js (hunk 1 above)
Comment (roberttod) [inline test/functional/test-queue-that-functional.js]: Until I fix the other tests, yes.

**Comment by roberttod** [comment]
Yeh, it is more complicated now unfortunately. I will add something to describe how it works or at least attempt to.

**Comment by davidpadbury** [comment]
Tested both normal behavior and retries after errors. Both scenarios worked as expected. Confirmed that no polling was occurring in local storage. Only verified in Chrome, but I assume unit tests are covering other browsers here.

**Comment by roberttod** [comment]
@rdecoene can I get a release not for this?

**Comment by davidpadbury** [comment]
let me just work out what's going on with the duplicate log messages for active queue switching I encountered while testing. seems a bit odd.

image

**Comment by davidpadbury** [comment]
So it's because we've slightly changed the active polling, but didn't update the active message. Looking the the `master` version there is a hard `switch` from a previously active queue to the current queue. Now we effectively run [this](https://github.com/QubitProducts/queue-that/blob/no-polling/lib/queue-that.js#L94-L95) block of code every time we process a batch in most cases "renewing" the active queue lease, or in exceptional circumstances switching. Perhaps logging to mirror that behavior would be helpful?

**Comment by davidpadbury** [comment]
(but either way, it's not technically a problem)

**Comment by roberttod** [comment]
You are spot on and I will update that message to only get logged when there is an actual switch happening. I was avoiding it because it needs some things to be moved around so that you don't require another read of localStorage, but these logs are pretty useful so good to ensure they make sense.

**Comment by rdecoene** [comment]
release notes live

**Pull request merged**